src/cmd/sanity_test/sanity_tests.go

```diff
@@ -363,27 +416,55 @@ func checkOnedriveRestoration(
 		// currently we don't restore blank folders.
 		// skip permission check for empty folders
 		if ptr.Val(driveItem.GetFolder().GetChildCount()) == 0 {
-			logger.Ctx(ctx).Info("skipped empty folder: ", itemName)
-			fmt.Println("skipped empty folder: ", itemName)
-
+			logAndPrint(ctx, "skipped empty folder: %s", itemName)
 			continue
 		}
 
-		folderPermission[itemName] = permissionIn(ctx, client, driveID, itemID)
-		getOneDriveChildFolder(ctx, client, driveID, itemID, itemName, fileSizes, folderPermission, startTime)
+		folderPermissions[itemName] = permissionIn(ctx, client, driveID, itemID)
```

Comment:
I know it's not directly related to this PR, but the fact that we're using file names here means we're making the assumption that file names are unique. We should have a note about that somewhere probably

Reply:
Could you kindly add that note/issue/whatever, please?